sdk/kiosk/src/tx/kiosk.ts

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  SuiAddress,
+  TransactionArgument,
+  TransactionBlock,
+} from '@mysten/sui.js';
+
+import { ObjectArgument, objArg } from '../utils';
+import { TransferRequest } from './transfer-request';
+
+/** The Kiosk module. */
+export const KIOSK_MODULE = '0x2::kiosk';
+
+/** The Kiosk type. */
+export const KIOSK_TYPE = `${KIOSK_MODULE}::Kiosk`;
+
+/**
+ * Create a new shared Kiosk and return the `KioskOwnerCap`
+ * and the transaction to continue building.
+ */
+export function createKiosk(
+  tx: TransactionBlock,
+): [TransactionArgument, TransactionArgument] {
+  let [kiosk, kioskOwnerCap] = tx.moveCall({
+    target: `${KIOSK_MODULE}::new`,
+  });
+
+  return [kiosk, kioskOwnerCap];
+}
+
+export function createKioskAndShare(tx: TransactionBlock): TransactionArgument {
+  let [kiosk, kioskOwnerCap] = tx.moveCall({
+    target: `${KIOSK_MODULE}::new`,
+  });
+
+  tx.moveCall({
+    target: `0x2::transfer::public_share_object`,
+    arguments: [kiosk],
+    typeArguments: [KIOSK_TYPE],
+  });
+
+  return kioskOwnerCap;
+}
+
+/**
+ * Call the `kiosk::place<T>(Kiosk, KioskOwnerCap, Item)` function.
+ * Place an item to the Kiosk.
+ */
+export function place(
+  tx: TransactionBlock,
+  itemType: string,
+  kiosk: ObjectArgument,
+  kioskCap: ObjectArgument,
+  item: ObjectArgument,
+) {
+  tx.moveCall({
+    target: `${KIOSK_MODULE}::place`,
+    typeArguments: [itemType],
+    arguments: [objArg(tx, kiosk), objArg(tx, kioskCap), objArg(tx, item)],
+  });
+
+  return tx;
```

Comment:
Seems like returning `tx` is inconsistent here, elsewhere it seems like you return either relevant TransactionArguments or void